pkg/netdevice/netDeviceProvider.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2020 Intel Corp. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package netdevice
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/golang/glog"
+	"github.com/intel/sriov-network-device-plugin/pkg/types"
+	"github.com/intel/sriov-network-device-plugin/pkg/utils"
+	"github.com/jaypipes/ghw"
+	"github.com/vishvananda/netlink"
+)
+
+type netDeviceProvider struct {
+	deviceList []types.PciNetDevice
+	rFactory   types.ResourceFactory
+}
+
+// NewNetDeviceProvider DeviceProvider implementation from netDeviceProvider instance
+func NewNetDeviceProvider(rf types.ResourceFactory) types.DeviceProvider {
+	return &netDeviceProvider{
+		rFactory:   rf,
+		deviceList: make([]types.PciNetDevice, 0),
```

Comment:
The zero len is redundant imo as no capacity is provided

Reply:
Cannot initialize a slice with make without giving its length.
Try here: https://play.golang.org/p/ae-ktK6cOLK

Reply:
My bad then, no idea why i remembered differently